rclrs/Cargo.toml

```diff
@@ -13,6 +16,10 @@ core-error = "0.0.0"
 parking_lot = {version = "0.11.2", optional = true}
 spin = "0.9.2"
 
+[dependencies.rclrs_common]
+version = "*"
+path = "../rclrs_common"
```

Comment:
Is there a way to avoid this? It'd be unfortunate because it's bypassing the ROS dependency resolution mechanism, but it makes `rust-analyzer`, it's fine as a compromise. Or maybe, it'd just make sense to integrate `rclrs_common` into `rclrs`. What do you think?

Reply:
I had been debating merging the two crates for some time. I will give that a shot and see how it goes.

I am doing something similar in `rclrs_examples` as well. I'm assuming it's a problem there as well? What I'd like to do is figure out some way to make ROS 2's dependency resolution work, while also allowing `cargo` the freedom to build crates as it likes, since that's what most IDEs will be expecting...

Reply:
Okay, that took quite a bit more effort than I was expecting, but I have managed to combine most of `rclrs_common` with `rclrs`. The message traits had to be spun off into their own crate, otherwise we ended up with circular build dependencies.

I can do a bit more thinking on how to avoid the explicit paths in the manifest. Perhaps we should use an auto-generated `.cargo/config.toml` file, such as the one that I use for the `std_msgs` dependencies?